lighthouse-core/fraggle-rock/gather/navigation-runner.js

```diff
@@ -224,8 +226,15 @@ async function _navigation(navigationContext) {
   await collectPhaseArtifacts({phase: 'startInstrumentation', ...phaseState});
   await collectPhaseArtifacts({phase: 'startSensitiveInstrumentation', ...phaseState});
   const navigateResult = await _navigate(navigationContext);
-  phaseState.baseArtifacts.URL.finalUrl = navigateResult.finalUrl;
-  phaseState.url = navigateResult.finalUrl;
+  if (navigationContext.isFirst) {
```

Comment:
could this boolean be removed if you check if `phaseState.baseArtifacts.URL` is defined instead?

Reply:
`phaseState.baseArtifacts.URL` should always be defined, but we could check if `initialUrl`/`finalUrl` are an empty string.